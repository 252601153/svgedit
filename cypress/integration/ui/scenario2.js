import {
  visitAndApproveStorage
} from '../../support/ui-test-helper.js';

const testSnapshot = () => {
  cy.get('#svgcontent').cleanSnapshot();
};

describe('use all parts of svg-edit', function () {
  before(() => {
    visitAndApproveStorage();
  });

  it('check tool_source_set', function () {
    cy.get('#tool_source').click({ force: true });
    cy.get('#svg_source_textarea')
      .type('{selectall}', { force: true })
      .type(`<svg width="640" height="480" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg">
      <g class="layer">
       <title>Layer 1</title>
       </g>
     </svg>`, { force: true, parseSpecialCharSequences: false });
    cy.get('#tool_source_save').click({ force: true });
    testSnapshot();
  });
  it('check tool_path', function () {
    cy.get('#tool_path')
      .click({ force: true });
    cy.get('#svgcontent')
      .trigger('mousedown', 50, 50, { force: true })
      .trigger('mouseup', { force: true })
      .trigger('mousemove', 100, 50, { force: true })
      .trigger('mousedown', 100, 50, { force: true })
      .trigger('mouseup', { force: true })
      .trigger('mousemove', 75, 150, { force: true })
      .trigger('mousedown', 75, 150, { force: true })
      .trigger('mouseup', { force: true })
      .trigger('mousemove', 0, 0, { force: true })
      .trigger('mousedown', 0, 0, { force: true })
      .trigger('mouseup', { force: true });
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_fhellipse', function () {
    cy.get('#tool_fhellipse')
      .click({ force: true });
    cy.get('#svgcontent')
      .trigger('mousedown', 200, 80, { force: true })
      .trigger('mousemove', 320, 80, { force: true })
      .trigger('mousemove', 320, 180, { force: true })
      .trigger('mousemove', 200, 180, { force: true })
      .trigger('mousemove', 200, 80, { force: true })
      .trigger('mouseup', 200, 80, { force: true });
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_ellipse_circle', function () {
    cy.get('#tool_ellipse').click({ force: true });
    cy.get('#svgcontent')
      .trigger('mousedown', 75, 150, { force: true })
      .trigger('mousemove', 130, 175, { force: true })
      .trigger('mouseup', { force: true });
    cy.get('#tool_circle').click({ force: true });
    cy.get('#svgcontent')
      .trigger('mousedown', 150, 50, { force: true })
      .trigger('mousemove', 150, 80, { force: true })
      .trigger('mouseup', { force: true });
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_circle_change_fill_color', function () {
    cy.get('#svg_2').click({ force: true });
    cy.get('#js-se-palette').find('.square').eq(8)
      .click({ force: true });
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_circle_change_opacity', function () {
    cy.get('#svg_2').click({ force: true });
    // cy.get('#opacity').trigger('click', 'bottomRight');
    for(let n = 0; n < 10; n ++){
      cy.get('#opacity').shadow().find('elix-number-spin-box').eq(0).shadow().find('#downButton').eq(0)
        .click({ force: true });
    }
    cy.get('#svgcontent').toMatchSnapshot();
  });
  it('check tool_ellipse_change_rotation', function () {
    cy.get('#svg_3').click({ force: true });
    // cy.get('#opacity').trigger('click', 'bottomRight');
    for(let n = 0; n < 5; n ++){
      cy.get('#angle').shadow().find('elix-number-spin-box').eq(0).shadow().find('#upButton').eq(0)
        .click({ force: true });
    }
    cy.get('#svgcontent').toMatchSnapshot();
  });

});
