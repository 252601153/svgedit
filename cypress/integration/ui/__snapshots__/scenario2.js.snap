exports[`use all parts of svg-edit > check tool_source_set #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer"><title>Layer 1</title></g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_circle #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      opacity="0.5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="1"
      ></animate>
    </circle>
  </g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_fhellipse #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      opacity="0.5"
      style="pointer-events:inherit"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="1"
      ></animate>
    </ellipse>
  </g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_ellipse #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    ></ellipse>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      opacity="0.5"
      style="pointer-events:inherit"
      cx="114"
      cy="189"
      rx="55"
      ry="25"
      id="svg_3"
    >
      <animate
        attributeName="opacity"
        begin="indefinite"
        dur="0.2"
        fill="freeze"
        to="1"
      ></animate>
    </ellipse>
  </g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_circle_change_fill_color #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
    ></ellipse>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="114"
      cy="189"
      rx="55"
      ry="25"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
    ></ellipse>
  </g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_circle_change_opacity #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
    ></ellipse>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="114"
      cy="189"
      rx="55"
      ry="25"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
    ></ellipse>
  </g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_ellipse_change_rotation #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
    ></ellipse>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      cx="114"
      cy="189"
      rx="55"
      ry="25"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
      transform="rotate(25 114 189)"
    ></ellipse>
  </g>
</svg>
`;

exports[`use all parts of svg-edit > check tool_ellipse_change_cx_cy_coordinate #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
    ></ellipse>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      cx="139"
      cy="214"
      rx="55"
      ry="25"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
      filter="url(#svg_3_blur)"
      transform="rotate(25 139 214)"
    ></ellipse>
  </g>
  <defs>
    <filter id="svg_3_blur">
      <feGaussianBlur in="SourceGraphic" stdDeviation="1"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_ellipse_change_rx_ry_radius #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
    ></ellipse>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      cx="139"
      cy="214"
      rx="80"
      ry="50"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
      filter="url(#svg_3_blur)"
      transform="rotate(25 139 214)"
    ></ellipse>
  </g>
  <defs>
    <filter id="svg_3_blur">
      <feGaussianBlur in="SourceGraphic" stdDeviation="1"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_ellipse_bring_to_back #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
    ></ellipse>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      cx="139"
      cy="214"
      rx="80"
      ry="50"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
      filter="url(#svg_3_blur)"
      transform="rotate(25 139 214)"
    ></ellipse>
  </g>
  <defs>
    <filter id="svg_3_blur">
      <feGaussianBlur in="SourceGraphic" stdDeviation="1"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_ellipse_bring_to_front #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      cx="139"
      cy="214"
      rx="80"
      ry="50"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
      filter="url(#svg_3_blur)"
      transform="rotate(25 139 214)"
    ></ellipse>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
    ></ellipse>
  </g>
  <defs>
    <filter id="svg_3_blur">
      <feGaussianBlur in="SourceGraphic" stdDeviation="1"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_ellipse_clone #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      cx="139"
      cy="214"
      rx="80"
      ry="50"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
      filter="url(#svg_3_blur)"
      transform="rotate(25 139 214)"
    ></ellipse>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
    ></ellipse>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="319"
      cy="189"
      rx="60"
      ry="50"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
      id="svg_4"
    ></ellipse>
  </g>
  <defs>
    <filter id="svg_3_blur">
      <feGaussianBlur in="SourceGraphic" stdDeviation="1"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;

exports[`use all parts of svg-edit > check tool_ellipse_change_blur #0`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:svg="http://www.w3.org/2000/svg"
  width="640"
  height="480"
  id="svgcontent"
  overflow="visible"
  x="640"
  y="480"
  viewBox="0 0 640 480"
>
  <g class="layer" style="pointer-events:all">
    <title style="pointer-events:inherit">Layer 1</title>
    <circle
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      style="pointer-events:inherit"
      cx="150"
      cy="150"
      r="111.80339887498948"
      id="svg_1"
      fill-opacity="1"
      stroke-opacity="1"
    ></circle>
    <ellipse
      fill="#ffff00"
      stroke="#000000"
      stroke-width="5"
      cx="299"
      cy="169"
      rx="60"
      ry="50"
      id="svg_2"
      fill-opacity="1"
      stroke-opacity="1"
      opacity="0.5"
    ></ellipse>
    <ellipse
      fill="#FF0000"
      stroke="#000000"
      stroke-width="5"
      cx="114"
      cy="189"
      rx="55"
      ry="25"
      id="svg_3"
      fill-opacity="1"
      stroke-opacity="1"
      transform="rotate(25 114 189)"
      filter="url(#svg_3_blur)"
    ></ellipse>
  </g>
  <defs>
    <filter id="svg_3_blur" x="-50%" y="-50%" width="200%" height="200%">
      <feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
    </filter>
  </defs>
</svg>
`;
